### PR TITLE
homework2 nazar.hlynskyi

### PR DESCRIPTION
**Pull request by nAzgalor**

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
Good job)
Small suggestions:
use getter/setter instead of instance variables
use hsh.fetch to provide defaults

File: lib/topics/state_and_behaviour.rb [hunk 1]
```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # basic object
+  class Car
+    attr_accessor :color, :model, :year, :current_speed
+
+    def initialize(params)
+      @color = params[:color] || 'color'
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: Would [keyword arguments](https://robots.thoughtbot.com/ruby-2-keyword-arguments) fit here?

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # basic object
+  class Car
+    attr_accessor :color, :model, :year, :current_speed
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: do you need to have public setters?

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -2,4 +2,27 @@
...
+
+    def initialize(params)
+      @color = params[:color] || 'color'
+      @model = params[:model] || 'model'
+      @year  = params[:year] || 2000
+      @current_speed = 0
+    end
+
+    def push_break(speed)
+      @current_speed -= speed if @current_speed >= speed
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb:21]: why don't you use setters here, if you already declared them?

File: lib/topics/state_and_behaviour.rb [hunk 4]
```diff
@@ -2,4 +2,27 @@
...
+    end
+
+    def speed_up(speed)
+      @current_speed += speed
+    end
+
+    def self.default_car
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb:28]: Please, read a style guide (methods ordering)

File: lib/topics/state_and_behaviour.rb [hunk 5]
```diff
@@ -2,4 +2,27 @@
...
+      Car.new
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: What is `self` inside this method?

**Review by oleksandr-kychun — APPROVED** [review]

**Pull request closed without merging**